### ISSUE TITLE
How to use nn.BatchNorm???

### ISSUE THREAD

**Issue by fmscole** [google/flax#2279]
My code is:
```

class CNN(nn.Module):
  """A simple CNN model."""

  @nn.compact
  def __call__(self, x):
    x = nn.Conv(features=32, kernel_size=(3, 3))(x)
    x = nn.BatchNorm(use_running_average=True)(x)
    x = nn.relu(x)
    x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
    x = nn.Conv(features=64, kernel_size=(3, 3))(x)
    x = nn.BatchNorm()(x,use_running_average=True)
    x = nn.relu(x)
    x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
    x = x.reshape((x.shape[0], -1))  # flatten
    x = nn.Dense(features=256)(x)
    x = nn.relu(x)
    x = nn.Dense(features=10)(x)
    return x
```

But ,get the errors:
```
~/anaconda3/lib/python3.8/site-packages/flax/core/scope.py in variable(self, col, name, init_fn, *init_args)
    716       if not self.is_mutable_collection(col) or init_fn is None:
    717         if self.is_collection_empty(col):
--> 718           raise errors.ScopeCollectionNotFound(col, name, self.path_text)
    719         raise errors.ScopeVariableNotFoundError(name, col, self.path_text)
    720       init_value = init_fn(*init_args)

ScopeCollectionNotFound: Tried to access "mean" from collection "batch_stats" in "/BatchNorm_0" but the collection is empty. (https://flax.readthedocs.io/en/latest/flax.errors.html#flax.errors.ScopeCollectionNotFound)
```
